### ISSUE TITLE
The values of `true` and `false` in functions of `vec`

### ISSUE THREAD

**Issue by Melirius** [KhronosGroup/SYCL-Docs#302]
In table 144 _Hidden friend functions of the vec class template_ of the spec for the functions at the end
``` cxx
vec<RET, NumElements> operatorOP(const vec& lhs, const vec &rhs); // Where OP is: ==, !=, <, >, <=, >=.
vec<RET, NumElements> operatorOP(const vec &lhs, const DataT &rhs); // Where OP is: ==, !=, <, >, <=, >=.
vec<RET, NumElements> operatorOP(const DataT &lhs, const vec &rhs); // Where OP is: ==, !=, <, >, <=, >=.
vec<RET, NumElements> operator!(const vec &v); 
```
it is stated that **(1)** _Each element of the SYCL vec that is returned must be -1 if the operation results in true and 0 if the operation results in false or this SYCL vec is a NaN._ and **(2)** _The DataT template parameter of the constructed SYCL vec, RET, varies depending on the DataT template parameter of this SYCL vec. For a SYCL vec with DataT of type int8_t or uint8_t RET must be int8_t. For a SYCL vec with DataT of type int16_t, uint16_t or half RET must be int16_t. For a SYCL vec with DataT of type int32_t, uint32_t or float RET must be int32_t. For a SYCL vec with DataT of type int64_t, uint64_t or double RET must be int64_t._

However, there are three another boolean type functions that state only (2). It is the cases of
``` cxx
vec<RET, NumElements> operatorOP(const vec &lhs, const vec &rhs); // Where OP is: &&, ||.
vec<RET, NumElements> operatorOP(const vec &lhs, const DataT &rhs); // Where OP is: &&, ||.
vec<RET, NumElements> operatorOP(const DataT &lhs, const vec &rhs); // Where OP is: &&, ||.
```

What is the specified output in the resulting vectors? Maybe (1) should be inserted into the description of these functions, as well as a mention that standard C++ rules are used for conversion of `vec` elements into `true` and `false` before applying these operations?

**Comment by gmlueck** [comment]
Yes, I agree.  That wording should be added.

**Comment by Melirius** [comment]
I'll prepare a PR then.

**Comment by keryell** [comment]
Yes, I think the idea for the `&&` and `||` operators were to return directly a `bool`. So there is a need for a conversion of `bool` before, which is what `&&` and `||` does in classic C++. But you are right, this should be clarified.

**Comment by AlexeySachkov** [comment]
Note that CTS is currently testing `operator&&`, `operator||` and `operator!` as any other relational operator, i.e. `true == -1`: [`tests/vector_operators/generate_vector_operators.py:477`](https://github.com/KhronosGroup/SYCL-CTS/blob/eec8701491114d1930c90679ba5072dd2a5162cb/tests/vector_operators/generate_vector_operators.py#L477)

Depending on the direction we take here, we may have to update CTS as well.

It also seems to me that @keryell and @gmlueck suggest different things - which one is the way to go?

**Comment by tomdeakin** [comment]
@gmlueck @keryell - can we discuss in WG call 11/9/23? Thanks,

**Comment by keryell** [comment]
To look at recent discussion about ISO C++ SIMD library.

**Comment by gmlueck** [comment]
> It also seems to me that @keryell and @gmlueck suggest different things - which one is the way to go?

I think these functions should return `-1` when the comparison is true and `0` when the comparison if false, which is the same as all the other `vec` comparison functions.  @keryell, are you suggesting that they should return something else?

**Comment by tomdeakin** [comment]
Next steps, implement this as a PR:
> I think these functions should return -1 when the comparison is true and 0 when the comparison if false, which is the same as all the other vec comparison functions.
